geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/index/HBaseFeatureIndex.scala

```diff
@@ -175,13 +175,15 @@ trait HBaseFeatureIndex extends HBaseFeatureIndexType
                            filter: HBaseFilterStrategyType,
                            hints: Hints,
                            ecql: Option[Filter],
-                           dedupe: Boolean): ScanConfig = {
+                           dedupe: Boolean,
+                           remote: Boolean): ScanConfig = {
 
     import HBaseFeatureIndex.{DataColumnFamily}
     import org.locationtech.geomesa.index.conf.QueryHints.RichHints
 
     /** This function is used to implement custom client filters for HBase **/
-      val toFeatures = resultsToFeatures(sft, ecql, hints.getTransform)
+      var toFeatures = resultsToFeatures(sft, ecql, hints.getTransform)
+      if (remote) {toFeatures = resultsToFeatures(sft, ecql, None)}
```

Comment:
instead of using mutatble state and calling the method twice (both things we try to avoid), do something like:
```
val transform = if (remote) { None } else { hints.getTransform }
val toFeatures = restultsToFeatures(sft, ecql, transform)